Sources/ProtobufferBuilder/EnrichedInfo.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  File.swift
+//  
+//
+//  Created by Nityananda on 11.12.20.
+//
+
+@_implementationOnly import Runtime
+
+struct EnrichedInfo {
+    enum Cardinality {
+        case zeroToOne
+        case exactlyOne
+        case zeroToMany
+    }
+    
+    let typeInfo: TypeInfo
+    let propertyInfo: PropertyInfo?
+    let propertiesOffset: Int?
+    
+    var cardinality: Cardinality = .exactlyOne
```

Comment:
what about optional array types, imo we should change `cardinality` to `[Cardinality]` to allow this.